File: frontend/types/unidade.ts
import { Reserva } from "./reservas";

export interface Unidade {
  id: string;
  nome: string;
  regiao: string;
  endereco: string;
  cep: string;
  CEP?: string;
  foto_url?: string | null;
  latitude: number;
  longitude: number;
  reservas?: Reserva[];
}

export interface UnidadeFormData {
  nome: string;
  regiao: string;
  cep: string;
  endereco: string;
  foto_url?: string;
  latitude?: number;
  longitude?: number;
}

File: backend/app/models/unidade.py
import uuid
from sqlalchemy import Column, String, Float
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.core.database import Base


class Unidade(Base):
    __tablename__ = "unidades"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)

    nome = Column(String, nullable=False)
    regiao = Column(String, nullable=False)
    cep = Column(String, nullable=False)
    endereco = Column(String, nullable=False)
    foto_url = Column(String, nullable=True)
    latitude = Column(Float, nullable=True)
    longitude = Column(Float, nullable=True)

    reservas = relationship("Reserva", back_populates="unidade")
